tools/conformance/generator.py

```diff
@@ -20,100 +19,111 @@
+
+        cmd = [args.decoder, input_file]
+        original_icc_filename = os.path.join(test_dir, 'original.icc')
+        reconstructed_filename = os.path.join(test_dir, 'reconstructed.jpg')
+        pixel_prefix = os.path.join(test_dir, 'reference')
+        cmd.extend(['-p', pixel_prefix])
+        cmd.extend(['-i', original_icc_filename])
+        cmd.extend(['-j', reconstructed_filename])
+        metadata_filename = os.path.join(test_dir, 'test.json')
+        cmd.extend(['-m', metadata_filename])
+
+        # Decode and generate the reference files.
+        subprocess.check_call(cmd)
+
+        with open(metadata_filename, 'r') as f:
+            metadata = json.load(f)
+
+        if os.path.exists(original_icc_filename):
+            metadata['original_icc'] = "original.icc"
+
+        if os.path.exists(reconstructed_filename):
+            metadata['reconstructed_jpeg'] = "reconstructed.jpg"
+
+        for frame in metadata['frames']:
+            frame['rms_error'] = args.rmse
+            frame['peak_error'] = args.peak_error
+
+        if 'preview' in metadata:
+            metadata['preview']['rms_error'] = args.rmse
+            metadata['preview']['peak_error'] = args.peak_error
+
+        # Create the test descriptor file.
+        with open(metadata_filename, 'w') as f:
+            json.dump(metadata, f, indent=2)
+
+    # Generate a corpus descriptor with the list of the all the test_id names,
+    # one per line.
+    with open(os.path.join(args.output, 'corpus.txt'), 'w') as f:
+        f.write(''.join(line + '\n' for line in test_ids))
 
 
 def main():
-  parser = argparse.ArgumentParser(description=__doc__)
-  parser.add_argument('--decoder', metavar='DECODER', required=True,
-                      help='path to the decoder binary under test.')
-  parser.add_argument('--output', metavar='DIR', required=True,
-                      help='path to the output directory')
-  parser.add_argument('inputs', metavar='JXL', nargs='+',
-                      help='path to input .jxl file(s)')
-  args = parser.parse_args()
-  GenerateConformanceCorpus(args)
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--decoder',
+                        metavar='DECODER',
+                        required=True,
+                        help='path to the decoder binary under test.')
+    parser.add_argument('--output',
+                        metavar='DIR',
+                        required=True,
+                        help='path to the output directory')
+    parser.add_argument('--peak_error',
+                        metavar='PEAK_ERROR',
+                        type=float,
+                        required=True,
+                        help='peak error for each testcase')
+    parser.add_argument('--rmse',
+                        metavar='RMSE',
+                        type=float,
+                        required=True,
+                        help='max RMSE for each testcase')
```

Comment:
shouldn't this be different per test-case?

Reply:
It could also be different per-frame - I think we can manually modify it when needed, and merge multiple corpora otherwise.